### DISCUSSION TITLE
Best way to handle multiprocessing issues

### DISCUSSION THREAD

**Discussion by pearcetm** [pyinstaller/pyinstaller#8330] [Help] [answered]
As described [here](https://github.com/zarr-developers/zarr-python/issues/1676), simply importing a package that uses multiprocessing was causing my simple script to launch itself continuously (only in the pyinstaller packaged application, running the script directly was fine).

Adding this to the top of my script fixed it:
```
import multiprocessing
multiprocessing.freeze_support()
```

But I'm wondering if there's a way in the spec file and/or command line to deal with this automatically. Or is adding this to the entry point for my project the best solution?

**Comment by bwoodsend** [comment, marked as answer]
It's multiprocessing which is very different from multithreading but no, calling `multiprocessing.freeze_support()` has to be done in code so that the parent process and child processes are able to identify themselves as such.

**Reply by pearcetm** [reply]
> It's multiprocessing which is very different from multithreading

Yes, thanks, I'll fix that typo for clarity.

> calling multiprocessing.freeze_support() has to be done in code so that the parent process and child processes are able to identify themselves as such.

Does this need to be done in the application code directly, or could it be added to a runtime hook or something like that? I'd like to make it as robust as possible.

**Reply by rokm** [reply]
Currently, you have to do it in application, as described in https://pyinstaller.org/en/stable/common-issues-and-pitfalls.html#multi-processing. 

In order to be able to handle this automatically from run-time hook, we will need to solve the problem of retrieving the code for the entry-point script, for cases when multi-processing is used from the entry-point script. This is a bit of a special case compared to modules, where code retrieval is not a problem.

**Reply by pearcetm** [reply]
Ok, thanks so much for the info!
